lib/truffle/truffle/lazy-rubygems.rb

```diff
@@ -0,0 +1,23 @@
+module Kernel
+  # Take this alias name so RubyGems will reuse this copy
+  # and skip the method below once RubyGems is loaded.
+  alias :gem_original_require :require
+
+  private def require(path)
+    begin
+      gem_original_require(path)
+    rescue LoadError
+      require 'rubygems'
+      require path
```

Comment:
No, because rubygems re-defines `require`. But yes, it would happen if a `require` while loading rubygems and before redefining `require` would fail.
There are ways to avoid it (one of the commit uses a global variable), but they become quite involved for synchronizing correctly in the case of two concurrent usages which trigger the loading of rubygems. I'd rather relies on the synchronization guaranties of `require` and make sure rubygems doesn't do too funky things while loading itself.